core/src/main/java/com/google/cloud/sql/core/SslSocketFactory.java

```diff
@@ -188,9 +190,26 @@ private static void logTestPropertyWarning(String property) {
   }
 
   private SSLSocket createAndConfigureSocket(
-      String instanceName, CertificateCaching certificateCaching) throws IOException {
+      String instanceName,
+      List<String> ipTypes,
+      CertificateCaching certificateCaching)
+      throws IOException {
     InstanceSslInfo instanceSslInfo = getInstanceSslInfo(instanceName, certificateCaching);
-    String ipAddress = instanceSslInfo.getInstanceIpAddress();
+    String ipAddress = getPreferredIp(ipTypes, instanceSslInfo);
+    if (ipAddress == null) {
```

Comment:
Instead of doing an explicit check for null here, why not just throw the error inside the getPrefferedIp function?